convert2rhel/unit_tests/checks_test.py

```diff
@@ -16,30 +16,401 @@
+
+import pytest
+
+from convert2rhel import checks, unit_tests
+from convert2rhel.checks import (
+    check_thirdparty_kmods,
+    ensure_compatibility_of_kmods,
+    get_host_kmods,
+    get_most_recent_unique_kernel_pkgs,
+    get_rhel_supported_kmods,
+    perform_final_checks_before_ponr,
+    perform_pre_checks,
+)
+from convert2rhel.toolopts import tool_opts
+from convert2rhel.unit_tests import GetLoggerMocked
+from convert2rhel.utils import run_subprocess
+
+
 try:
     import unittest2 as unittest  # Python 2.6 support
 except ImportError:
     import unittest
 
-from convert2rhel import checks, unit_tests
-from convert2rhel.unit_tests import GetLoggerMocked
 
+try:
+    import unittest2 as unittest  # Python 2.6 support
+except ImportError:
+    import unittest
+
+
+if sys.version_info[:2] <= (2, 7):
+    import mock  # pylint: disable=import-error
+else:
+    from unittest import mock  # pylint: disable=no-name-in-module
+
+
+HOST_MODULES_STUB_GOOD = (
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/a.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/b.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/c.ko.xz\n"
+)
+HOST_MODULES_STUB_BAD = (
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/d.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/e.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/f.ko.xz\n"
+)
+REPOQUERY_F_STUB_GOOD = (
+    b"kernel-core-0:4.18.0-240.10.1.el8_3.x86_64\n"
+    b"kernel-core-0:4.18.0-240.15.1.el8_3.x86_64\n"
+    b"kernel-debug-core-0:4.18.0-240.10.1.el8_3.x86_64\n"
+    b"kernel-debug-core-0:4.18.0-240.15.1.el8_3.x86_64\n"
+)
+REPOQUERY_F_STUB_BAD = (
+    b"kernel-idontexpectyou-core-sdsdsd.ell8_3.x86_64\n"
+    b"kernel-core-0:4.18.0-240.15.1.el8_3.x86_64\n"
+    b"kernel-debug-core-0:4.18.0-240.10.1.el8_3.x86_64\n"
+    b"kernel-debug-core-0:4.18.0-240.15.1.el8_3.x86_64\n"
+)
+REPOQUERY_L_STUB_GOOD = (
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/a.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/a.ko\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/b.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/c.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/c.ko\n"
+)
+REPOQUERY_L_STUB_BAD = (
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/d.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/d.ko\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/e.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/f.ko.xz\n"
+    b"/lib/modules/5.8.0-7642-generic/kernel/lib/f.ko\n"
+)
+
+
+def test_checks_entrypoints(monkeypatch):
+    ensure_compatibility_of_kmods_mock = mock.Mock()
+    check_uefi_mock = mock.Mock()
+    monkeypatch.setattr(
+        checks,
+        "ensure_compatibility_of_kmods",
+        value=ensure_compatibility_of_kmods_mock,
+    )
+    monkeypatch.setattr(
+        checks,
+        "check_uefi",
+        value=check_uefi_mock,
+    )
+    perform_final_checks_before_ponr()
+    perform_pre_checks()
```

Comment:
Tests shouldn't be testing two methods at the same time, should split into two tests